component/module/file/file_test.go

```diff
@@ -93,12 +95,20 @@ func TestMissingFile(t *testing.T) {
 		DataPath:      t.TempDir(),
 	}
 
-	cfg := `
-		filename = "` + filepath.Join(t.TempDir(), "module.river") + `"
-	`
+	filePath := filepath.Join(t.TempDir(), "module.river")
+	cfg := `filename = "` + riverEscape(filePath) + `"`
+
 	var args Arguments
 	require.NoError(t, river.Unmarshal([]byte(cfg), &args))
 
 	_, err := New(opts, args)
 	require.ErrorContains(t, err, "failed to read file:")
 }
+
+func riverEscape(filePath string) string {
+	if runtime.GOOS == "windows" {
+		return strings.Replace(filePath, `\`, `\\`, -1)
+	}
+
+	return filePath
+}
```

Comment:
I think you can use filepath.FromSlash here instead of defining a new function.

Reply:
I'm open to suggestion on a better way but I'm specifically escaping the filepath for putting it into the river config. The suggested function I don't think is quite what I need to do the transform from `C:\Users\erikr\AppData\Local\Temp\TestMissingFile1834357519\002\module.river` to `C:\\Users\\erikr\\AppData\\Local\\Temp\\TestMissingFile1834357519\\002\\module.river`

(I did try it just to sanity check)

```golang
// FromSlash returns the result of replacing each slash ('/') character
// in path with a separator character. Multiple slashes are replaced
// by multiple separators.
func FromSlash(path string) string {
	if Separator == '/' {
		return path
	}
	return strings.ReplaceAll(path, "/", string(Separator))
}
```

Reply:
I thought about running the replace on all operating systems then chose to just mess with windows. I could also do the replace up above alternatively across all operating systems and not need a helper function but it may not be as clear as to why I am doing it.

Reply:
Doh, sorry, I didn't think it through enough. You could use `filepath.ToSlash` here but this is fine.